src/main/java/org/zalando/nakadi/service/subscription/SubscriptionValidationService.java

```diff
@@ -101,6 +103,9 @@ private void validateInitialCursors(final SubscriptionBase subscription,
         try {
             for (final SubscriptionCursorWithoutToken cursor : subscription.getInitialCursors()) {
                 final NakadiCursor nakadiCursor = cursorConverter.convert(cursor);
+                if (nakadiCursor.getTimeline().isDeleted()) {
+                    throw new InvalidCursorException(UNAVAILABLE, nakadiCursor);
```

Comment:
Will not it result in 422 when providing expired cursor? I understand that now it's even worse (probably the returned status code is 500) but I think it's an important use-case when creating a subscription based on cursors of another subscription. It can be that one of event-types of old subscription has latest cursors committed in some old timeline.

Reply:
@v-stepanov If I understood correctly - we decided that it is a way to go now - not allowed to put cursors that are not available.